stdlib/public/core/Codable.swift

```diff
@@ -3574,10 +3574,11 @@ public extension KeyedEncodingContainerProtocol {
 
 // Default implementations for decode(_:forKey:) in terms of decodeIfPresent(_:forKey:)
 public extension KeyedDecodingContainerProtocol {
-    public func decode(_ type: Bool.Type, forKey key: Key) throws -> Bool {
-        if let value = try decodeIfPresent(Bool.self, forKey: key) {
-            return value
-        } else if contains(key) {
+    @_semantics("optimize.sil.specialize.generic.never")
+    @_semantics("optimize.sil.specialize.generic.partial.never")
+    @inline(never)
+    internal func decodeError<T: Decodable>(_ type: T.Type, forKey key: Key) throws -> T {
```

Comment:
What's the point of this throwing the error instead of just returning a new error?
Can't this just be `internal func _errorInDecoding(_ type: T.Type, forKey key: Key) -> DecodingError { ... }`